From 0eff9b631c33c742394a2a8d444e40b2ed67e6eb Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 16 Jan 2026 14:09:36 +1100
Subject: [PATCH 05/10] default to CPU as device on osx-64, remove metal

Metal is only supported for M1+ devices, see
https://github.com/tinygrad/tinygrad/blob/v0.11.0/docs/runtime.md?plain=1#L10
and also https://github.com/tinygrad/tinygrad/issues/13866

sort CPU first to avoid trying things not necessarily supported on osx-64
---
 tinygrad/device.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tinygrad/device.py b/tinygrad/device.py
index efbe8ed9b..3b0fbcce9 100644
--- a/tinygrad/device.py
+++ b/tinygrad/device.py
@@ -11,7 +11,7 @@ from tinygrad.renderer import Renderer
 
 # **************** Device ****************
 
-ALL_DEVICES = ["METAL", "AMD", "NV", "CUDA", "QCOM", "CL", "CPU", "DSP", "WEBGPU"]
+ALL_DEVICES = ["CPU", "AMD", "NV", "CUDA", "QCOM", "CL", "DSP", "WEBGPU"]
 class _Device:
   def __init__(self) -> None:
     self._devices = [x.stem[len("ops_"):].upper() for x in (pathlib.Path(__file__).parent/"runtime").iterdir() if x.stem.startswith("ops_")]
